.circleci/config.yml

```diff
@@ -1,277 +1,85 @@
-# Javascript Node CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-javascript/ for more details
-#
 version: 2
+
+defaults: &defaults
+  working_directory: ~/project
+  docker:
+    - image: circleci/node:10
+
+attach: &attach
+  attach_workspace:
+    at: ~/project
+
 jobs:
   Build:
-    docker: [ { image: 'circleci/node:8' } ]
+    <<: *defaults
     steps:
       - checkout
-      - run: cat ./packages/apollo-link/package.json ./packages/apollo-link-batch/package.json ./packages/apollo-link-batch-http/package.json ./packages/apollo-link-context/package.json ./packages/apollo-link-dedup/package.json ./packages/apollo-link-error/package.json ./packages/apollo-link-http/package.json ./packages/apollo-link-http-common/package.json ./packages/apollo-link-polling/package.json ./packages/apollo-link-retry/package.json ./packages/apollo-link-schema/package.json ./packages/apollo-link-ws/package.json ./packages/zen-observable-ts/package.json > package-checksum
       - restore_cache:
           keys:
-            - dependency-cache-{{ checksum "package.json" }}-{{ checksum "package-checksum" }}
-            - dependency-cache-{{ checksum "package.json" }}
+            - npm-v2-{{ .Branch }}-{{ checksum "package-lock.json" }}
+            - npm-v2-{{ .Branch }}-
+            - npm-v2-
       - run:
           name: install-npm
           command: npm install
-      - run: git checkout package.json
       - save_cache:
-          key: dependency-cache-{{ checksum "package.json" }}-{{ checksum "package-checksum" }}
+          key: npm-v2-{{ .Branch }}-{{ checksum "package-lock.json" }}
           paths:
-            - ./node_modules
-            - ./packages/apollo-link/node_modules
-            - ./packages/apollo-link-batch/node_modules
-            - ./packages/apollo-link-batch-http/node_modules
-            - ./packages/apollo-link-context/node_modules
-            - ./packages/apollo-link-dedup/node_modules
-            - ./packages/apollo-link-error/node_modules
-            - ./packages/apollo-link-http/node_modules
-            - ./packages/apollo-link-http-common/node_modules
-            - ./packages/apollo-link-polling/node_modules
-            - ./packages/apollo-link-retry/node_modules
-            - ./packages/apollo-link-schema/node_modules
-            - ./packages/apollo-link-ws/node_modules
-            - ./packages/zen-observable-ts/node_modules
+            - ~/.npm
       - persist_to_workspace:
           root: .
           paths:
             - .
 
   Danger:
-    docker: [ { image: 'circleci/node:8' } ]
+    <<: *defaults
     steps:
-      - attach_workspace:
-          at: .
+      - <<: *attach
       - run: npm run danger
 
   Prettier Check:
-    docker: [ { image: 'circleci/node:8' } ]
+    <<: *defaults
     steps:
-      - attach_workspace:
-          at: .
+      - <<: *attach
       - run: npm run lint-check
 
   Docs:
```

Comment:
I know we're building the docs like this in other repos, but I have a feeling this isn't really necessary. Our Netlify based docs build and deploy process is already taking care of this, and if the docs fail to build for any reason, an  error will be reported by the Netlify GitHub check. Let's leave this docs CI step for now, but food for future thought.